.github/workflows/_e2e_tests.yaml

```diff
@@ -70,22 +70,30 @@ jobs:
       - name: Print test names
         run: echo "Test names ${{ steps.set_test_names.outputs.test_names }}"
 
-  # TODO: extract this as a separate workflow reading the dependencies versions from a file (not only Kind, but also
-  # GKE, Istio, Kong, etc.)
-  # https://github.com/Kong/kubernetes-ingress-controller/issues/3349
   dependencies-versions:
     runs-on: ubuntu-latest
     outputs:
-      kind-versions: ${{ steps.set-kind-versions.outputs.kind-versions }}
+      kind: ${{ steps.set-versions.outputs.kind }}
+      gke: ${{ steps.set-versions.outputs.gke }}
+      istio: ${{ steps.set-versions.outputs.istio }}
     steps:
-      - id: set-kind-versions
-        name: Set Kind versions
+      - name: checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - id: set-versions
+        name: Set versions
         run: |
           if [ "${{ inputs.all-supported-k8s-versions }}" == "true" ]; then
-            echo "kind-versions=[\"1.23.17\", \"1.24.15\", \"1.25.11\", \"1.26.6\", \"1.27.3\", \"1.28.0\"]" >> $GITHUB_OUTPUT
+            echo "kind=$(cat .github/test_dependencies.yaml | yq -ojson '.e2e.kind' | jq -c)" >> $GITHUB_OUTPUT
           else
-            echo "kind-versions=[\"1.28.0\"]" >> $GITHUB_OUTPUT
+            # We assume the first version in the list is the latest one.
+            echo "kind=$(cat .github/test_dependencies.yaml | yq -ojson '.e2e.kind' | jq -c '[first]')" >> $GITHUB_OUTPUT
```

Comment:
`jq -c '[first]'`

Where is this described in `jq` docs? I cannot find a mention of it nor can I make this work on my machine

```
curl -s 172.18.0.101:8001 | jq '.configuration.proxy_listeners'
[
  {
    "ip": "0.0.0.0",
    "ssl": false,
    "bind": false,
    "http2": false,
    "proxy_protocol": false,
    "deferred": false,
    "so_keepalive=off": false,
    "reuseport": false,
    "backlog=%d+": false,
    "ipv6only=on": false,
    "listener": "0.0.0.0:8000",
    "so_keepalive=on": false,
    "port": 8000,
    "so_keepalive=%w*:%w*:%d*": false,
    "ipv6only=off": false
  },
  {
    "ip": "0.0.0.0",
    "ssl": true,
    "bind": false,
    "http2": true,
    "proxy_protocol": false,
    "deferred": false,
    "so_keepalive=off": false,
    "reuseport": false,
    "backlog=%d+": false,
    "ipv6only=on": false,
    "listener": "0.0.0.0:8443 ssl http2",
    "so_keepalive=on": false,
    "port": 8443,
    "so_keepalive=%w*:%w*:%d*": false,
    "ipv6only=off": false
  }
]
```

```
curl -s 172.18.0.101:8001 | jq '.configuration.proxy_listeners[first]'
jq: error (at <stdin>:0): Cannot index object with number
```

Can't we just use `[0]`? 🤔

Reply:
As I understand, `first` is a sugar syntax for `.[0]`. An equivalent of `cat .github/test_dependencies.yaml | yq -ojson '.e2e.kind' | jq -c '[first]'` would be `cat .github/test_dependencies.yaml | yq -ojson '.e2e.kind' | jq -c '[.[0]]'`. 

Please note that the outer square brackets are for building an array with one element - `first` (or `.[0]`).

To make your example work, it would have be written this way: `curl -s 172.18.0.101:8001 | jq '.configuration.proxy_listeners | first'`

Reply:
It's documented [here](https://jqlang.github.io/jq/manual/#first-last-nth-1), but I found it mentioned somewhere on SO.

Reply:
Oh ok, nifty but hacky :shipit: